packages/block-library/src/gallery/edit.js

```diff
@@ -334,6 +340,9 @@ function GalleryEdit( props ) {
 			onError={ onUploadError }
 			notices={ hasImages ? undefined : noticeUI }
 			onFocus={ onFocus }
+			autoOpenMediaUpload={
```

Comment:
I am wondering if there's a way to make it more obvious that this functionality is only for mobile. I am thinking a comment might work here if we aren't able to extract this type of logic in any way. Let me know what you think. My concern is, I wouldn't want someone to come into this component and not understand what's being triggered because if they look at the MediaPlaceholder for the web this prop isn't being used there.

Reply:
That's a good question. I understand your concern that a comment might not go far enough given that while `gallery/edit.js` is one file for both platforms, `MediaPlaceholder` has platform-specific implementations. There might be precedent for this though, passing a platform-specific prop from a shared component (such as Gallery) to a component that has platform-specific implementations (like `MediaPlaceholder`).

I do see that `MediaPlaceholder` does have web-specific props, e.g. `addToGallery`:
https://github.com/WordPress/gutenberg/blame/1337bb39f8bae07fcf3fa7b8fafc9ee4d5f9a81d/packages/block-editor/src/components/media-placeholder/README.md#L44-L52

So I suppose documentation of the new prop might be enough.